src/java/htsjdk/samtools/MergingSamRecordIterator.java

```diff
@@ -144,17 +145,6 @@ public SAMRecord next() {
             }
         }
 
-        // Fix up the sequence indexes if needs be
-        if (this.samHeaderMerger.hasMergedSequenceDictionary()) {
-            if (record.getReferenceIndex() != SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
-                record.setReferenceIndex(this.samHeaderMerger.getMergedSequenceIndex(iterator.getReader().getFileHeader(), record.getReferenceIndex()));
-            }
-
-            if (record.getReadPairedFlag() && record.getMateReferenceIndex() != SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
-                record.setMateReferenceIndex(this.samHeaderMerger.getMergedSequenceIndex(iterator.getReader().getFileHeader(), record.getMateReferenceIndex()));
-            }
-        }
```

Comment:
Are we 100% sure that we can delete this block? I know that the header merger uses per-header translation tables mapping from old to new indices -- does the `record.setHeader(this.samHeaderMerger.getMergedHeader());` line above render this unnecessary?